subprojects/gst-plugins-bad/sys/msdk/gstmsdkdec.c

```diff
@@ -286,6 +289,73 @@ get_surface (GstMsdkDec * thiz, GstBuffer * buffer)
   return NULL;
 }
 
+static GstMsdkSurface *
+allocate_output_surface (GstMsdkDec * thiz)
+{
+  GstMsdkSurface *msdk_surface = NULL;
+  GstBuffer *out_buffer = NULL;
+
+  /* Free un-unsed msdk surfaces firstly, hence the associated mfx
+   * surfaces will be moved from used list to available list */
+  gst_msdkdec_free_unlocked_msdk_surfaces (thiz);
+#ifndef _WIN32
+  if ((gst_buffer_pool_acquire_buffer (thiz->alloc_pool, &out_buffer, 0))
+      != GST_FLOW_OK) {
+    GST_ERROR_OBJECT (thiz, "Failed to allocate output buffer");
+    return NULL;
+  }
+
+  GstMemory *mem = NULL;
+  mfxFrameSurface1 *mfx_surface = NULL;
+
+  mem = gst_buffer_peek_memory (out_buffer, 0);
+  msdk_surface = g_slice_new0 (GstMsdkSurface);
+  msdk_surface->from_qdata = FALSE;
+
+  if ((mfx_surface = gst_mini_object_get_qdata (GST_MINI_OBJECT_CAST (mem),
+              GST_MSDK_FRAME_SURFACE))) {
+    msdk_surface->surface = mfx_surface;
+    msdk_surface->from_qdata = TRUE;
+  }
```

Comment:
Adding an else to handle mfx_surface == NULL and remove line 344 might be better

Reply:
> Adding an else to handle mfx_surface == NULL and remove line 344 might be better

Thanks for the advice. Done.